### DISCUSSION TITLE
Cloning of a Method?

### DISCUSSION THREAD

**Discussion by 7enderhead** [0xd4d/dnlib#493] [Q&A] [answered]
- Is there a canonical way to "clone" a method?
- I have a `MethodDefMD` and would like to have a clone (with a different name of course), whose body's instructions I would then change/adapt.
- I can create a `MethodDefUser`, using the `MethodSig` of the original method, but I see no way to, e.g., change the new method's parameter names (which are all auto-generated like `A_0`, `A_1` etc).

**Comment by UlyssesWu** [comment, marked as answer]
> but I see no way to, e.g., change the new method's parameter names`

`MethodDef.ParamDefs` & `new ParamDefUser`

**Reply by 7enderhead** [reply]
Great, thanks!
